### PR TITLE
0.13 docs update: terraform-plugin

### PR DESCRIPTION
**Pull request by Walther**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Related to https://github.com/garden-io/garden/issues/4067

**Special notes for your reviewer**:

BREAKING CHANGE: `garden plugins terraform apply-module` is now `garden plugins terraform apply-deploy`, and similar change for the `plan-*` and `destroy-*` commands.

(Let me know if we want to use some other name for these instead, e.g. `terraform apply-action` or something)

### CODE REVIEW

File: docs/terraform-plugin/about.md [hunk 1]
```diff
@@ -11,13 +11,13 @@ It's strongly recommended that you [learn about Terraform](https://www.terraform
 
 ## How it works
 
-Under the hood, Garden simply wraps Terraform, so there's no magic involved. Garden just automates its execution and makes stack outputs available to your Garden providers and modules.
+Under the hood, Garden simply wraps Terraform, so there's no magic involved. Garden just automates its execution and makes stack outputs available to your Garden providers and actions.
 
-The `terraform` provider can both provision a Terraform stack when initializing Garden, or through `terraform` modules that are deployed like other services in your stack.
+The `terraform` provider can both provision a Terraform stack when initializing Garden, or through `terraform` actions that are utilized like other actions in your stack.
```
Comment (Orzelius) [inline docs/terraform-plugin/about.md:16]:
```suggestion
Terraform resources can be provisioned through the `terraform` provider when initializing Garden, or via `terraform` actions that are utilized like other actions in your stack.
```

I think this goes smoother

File: docs/terraform-plugin/about.md [hunk 2]
```diff
@@ -11,13 +11,13 @@ It's strongly recommended that you [learn about Terraform](https://www.terraform
...
 
-The latter method, using one or more `terraform` _modules_, can be better if your other providers don't need to reference the stack outputs but your _services, tasks and tests_ do. In this style, you can basically create small Terraform stacks that are part of your Stack Graph much like other services. A good example would be deploying a database instance, that other services in your project can then connect to.
+The latter method, using one or more `terraform` _Deploy actions_, can be better if your other providers don't need to reference the stack outputs but your other Deploy, Run and Test actions do. In this style, you can basically create small Terraform stacks that are part of your Stack Graph much like other services. A good example would be deploying a database instance, that other services in your project can then connect to.
```
Comment (Orzelius) [inline docs/terraform-plugin/about.md:20]:
```suggestion
Using `terraform` _Deploy actions_, can be better if your other providers don't need to reference the stack outputs but other Deploy, Run and Test actions do. In this style, you can basically create small Terraform stacks that are part of your Stack Graph much like other services. A good example would be deploying a database instance, that other services in your project can then connect to.
```

Simplification

File: docs/terraform-plugin/about.md [hunk 3]
```diff
@@ -32,21 +32,21 @@ We only recommend using `autoApply` for private development environments, since
 To manually plan and apply stacks, we provide the following commands:
 
 ```console
-garden --env=<env-name> plugins terraform apply-root                  # Runs `terraform apply` for the provider root stack.
-garden --env=<env-name> plugins terraform apply-module -- <module-name>  # Runs `terraform apply` for the specified terraform module.
-garden --env=<env-name> plugins terraform plan-root                   # Runs `terraform plan` for the provider root stack.
-garden --env=<env-name> plugins terraform plan-module -- <module-name>   # Runs `terraform plan` for the specified terraform module.
+garden --env=<env-name> plugins terraform apply-root                     # Runs `terraform apply` for the provider root stack.
+garden --env=<env-name> plugins terraform apply-deploy -- <action-name>  # Runs `terraform apply` for the specified terraform Deploy action.
```
Comment (Orzelius) [inline docs/terraform-plugin/about.md:36]: I think this should be `apply-action` after all, kinda just sounds smoother grammar wise

**Review by Orzelius — CHANGES_REQUESTED** [review]

File: docs/terraform-plugin/about.md (hunk 3 above)
Comment (Walther) [reply docs/terraform-plugin/about.md]: Sounds good, changed in [d2b2abd](https://github.com/garden-io/garden/pull/4296/commits/d2b2abd24312be9f1ca59fca8d3c0e4d773632ac)

**Review by Orzelius — APPROVED** [review]

**Pull request merged**